<!DOCTYPE html>
<html>
<head>
<title>2-12. Using a rejection handler at the end of a chain</title>
<link href="../prism.css" rel="stylesheet">
<script src="../prism.js"></script>

<script>
Promise.reject(Error('bad news')).then(
	function step2() {
		console.log('This is never run');
	}
).then(
	function step3() {
		console.log('This is also never run');
	}
).catch(
	function (error) {
		console.log('Something failed along the way. Inspect error for more info.');
		console.log(error); // Error object with message: 'bad news'
	}
);
</script>

</head>
<body>

<nav>
<a href="2-11-callback-execution-order.html">Previous</a> | <a href="2-13-reject-using-throw.html">Next</a>
</nav>

<h1>Example 2-12. Using a rejection handler at the end of a chain</h1>

<pre>
<code class="language-javascript">
Promise.reject(Error('bad news')).then(
	function step2() {
		console.log('This is never run');
	}
).then(
	function step3() {
		console.log('This is also never run');
	}
).catch(
	function (error) {
		console.log('Something failed along the way. Inspect error for more info.');
		console.log(error); // Error object with message: 'bad news'
	}
);

// Console output:
// Something failed along the way. Inspect error for more info.
// [Error object] { message: 'bad news' ... }
</code>
</pre>

</body>
</html>
